judge/views/user.py

```diff
@@ -507,3 +512,106 @@ def post(self, request, *args, **kwargs):
             return HttpResponse(_('You have sent too many password reset requests. Please try again later.'),
                                 content_type='text/plain', status=429)
         return super().post(request, *args, **kwargs)
+
+
+class EmailChangeRequestView(LoginRequiredMixin, TitleMixin, FormView):
+    title = _('Change your email')
+    template_name = 'registration/email_change.html'
+    form_class = EmailChangeForm
+
+    activate_html_email_template_name = 'registration/email_change_activate_email.html'
+    activate_email_template_name = 'registration/email_change_activate_email.txt'
+    activate_subject_template_name = 'registration/email_change_activate_subject.txt'
+    notify_html_email_template_name = 'registration/email_change_notify_email.html'
+    notify_email_template_name = 'registration/email_change_notify_email.txt'
+    notify_subject_template_name = 'registration/email_change_notify_subject.txt'
+
+    def form_valid(self, form):
+        signer = signing.TimestampSigner()
+        new_email = form.cleaned_data['email']
+        activation_key = base64.urlsafe_b64encode(signer.sign_object({
+            'id': self.request.user.id,
+            'email': new_email,
+        }).encode()).decode()
+
+        current_site = get_current_site(self.request)
+        context = {
+            'domain': current_site.domain,
+            'site_name': current_site.name,
+            'protocol': 'https' if self.request.is_secure() else 'http',
+            'site_admin_email': settings.SITE_ADMIN_EMAIL,
+            'expiry_minutes': settings.DMOJ_EMAIL_CHANGE_EXPIRY_MINUTES,
+            'user': self.request.user,
+            'activation_key': activation_key,
+            'new_email': new_email,
+        }
+        # When from_email is none, we use the DEFAULT_FROM_EMAIL setting
+        send_mail(
+            self.notify_subject_template_name, self.notify_email_template_name, context, from_email=None,
```

Comment:
If you are going to pass these as kwargs, then you should make it explicitly required to do so by adding `*` to the parameter list of `send_mail`.